lib/src/action.dart

```diff
@@ -41,7 +43,7 @@ import 'dart:async';
 /// when a consumer needs to check state changes immediately after invoking an
 /// action.
 ///
-class Action<T> implements Function {
```

Comment:
Mixing it in gets us all the helper methods for free, that's the ideal way to use `Disposable`. Does that answer your question?